exec.py

```diff
@@ -124,47 +145,139 @@ def plot_plane(ws, tag):
 
     fig.savefig(f"output/test_{tag}.png", dpi=300)
 
+def divide_into_chunks(args):
+    '''Divide the number of events in the input file into given chunk sizes.'''
+    # Read the input file
+    infile = uproot.open(args.inpath)
 
-def main():
+    eventchunksize = args.chunksize
+
+    nevents = len(infile['Events'])
+    nchunks = nevents // eventchunksize + 1
+
+    event_chunks = []
+
+    for idx in range(nchunks-1):
+        event_chunks.append(
+            range(eventchunksize*idx, eventchunksize*(idx+1))
+        )
+
+    # The last chunk
+    remainder = nevents % eventchunksize
+    event_chunks.append(
+        range(eventchunksize*(nchunks-1), eventchunksize*(nchunks-1) + remainder)
+    )
+
+    return event_chunks
+
+def run_chunk(event_chunk, nchunk, outdir, logdir, args, do_plot=False, do_print=False):
```

Comment:
I'd suggest to remove all plotting from this script. The plane plots were a nice per-event debugging tool to get this fit started, but if you are doing serious processing, you are not going to look at 10^N plots. Suggest to move all plotting into a separate script so that it is still easy to make plots for a given workspace if needed.